src/background/transactions/TransactionService.ts

```diff
@@ -52,22 +53,38 @@ export class TransactionService {
     this.addListeners();
   }
 
+  private markAsDropped(item: TransactionObject | undefined) {
+    if (item) {
+      this.transactionsStore.upsertTransaction({ ...item, dropped: true });
+    }
+  }
+
   addListeners() {
-    emitter.on('transactionSent', ({ transaction, initiator }) => {
-      const newItem = {
-        transaction,
-        hash: transaction.hash,
-        initiator,
-        timestamp: Date.now(),
-      };
-      this.transactionsPoller.add([toPollingObj(newItem)]);
+    emitter.on(
+      'transactionSent',
+      ({ transaction, initiator, addressAction }) => {
+        const newItem: TransactionObject = {
+          transaction,
+          hash: transaction.hash,
+          initiator,
+          timestamp: Date.now(),
+        };
+        if (
+          addressAction &&
+          isLocalAddressAction(addressAction) &&
+          addressAction.relatedTransaction
+        ) {
+          newItem.relatedTransactionHash = addressAction.relatedTransaction;
```

Comment:
A bit confusing that transactionHash = transaction

Reply:
Since we're augmenting the backend entity `AddressAction`, I tried to come up with a name that they're more likely to introduce on the backend if they ever do it. But locally I wanted to have a less confusing name.